proposals/dao/vip_12.ts

```diff
@@ -0,0 +1,148 @@
+      decimalsNormalizer: 0,
+      doInvert: true, /// invert the price so that the Oracle and PSM works correctly,
+      volt: voltV2.address
+    },
+    0, // 0 mint fee
+    0, // 0 redeem fee
+    reservesThreshold,
+    mintLimitPerSecond,
+    voltPSMBufferCap,
+    addresses.dai,
+    addressOne
+  );
+
+  await daiPriceBoundPSM.deployed();
+
+  console.log(`\nDAI PriceBoundPSM deployed to: ${daiPriceBoundPSM.address}`);
+
+  const usdcPriceBoundPSM = await (
+    await ethers.getContractFactory('PriceBoundPSM')
+  ).deploy(
+    voltUsdcFloorPrice,
+    voltUsdcCeilingPrice,
+    {
+      coreAddress: addresses.core,
+      oracleAddress: addresses.voltSystemOraclePassThrough, // OPT
+      backupOracle: ethers.constants.AddressZero,
+      decimalsNormalizer: 12,
+      doInvert: true, /// invert the price so that the Oracle and PSM works correctly,
+      volt: voltV2.address
+    },
+    0,
+    0,
+    reservesThreshold,
+    mintLimitPerSecond,
+    voltPSMBufferCap,
+    addresses.usdc,
+    addressOne
+  );
+
+  console.log(`\nUSDC PriceBoundPSM deployed to: ${usdcPriceBoundPSM.address}`);
+
+  const TempCoreRef = await ethers.getContractFactory('TempCoreRef');
+  const tempCoreRef = await TempCoreRef.deploy(addresses.core, voltV2.address);
+  await tempCoreRef.deployed();
+
+  console.log(`\nTempCoreRef deployed to: ${tempCoreRef.address}`);
+
+  const VoltMigrator = await ethers.getContractFactory('VoltMigrator');
+  const voltMigrator = await VoltMigrator.deploy(addresses.core, voltV2.address);
+  await voltMigrator.deployed();
+
+  console.log(`\nVolt Migrator deployed to: ${voltMigrator.address}`);
+
+  const MigratorRouter = await ethers.getContractFactory('MigratorRouter');
+  const migratorRouter = await MigratorRouter.deploy(
+    addresses.core,
+    voltV2.address,
+    daiPriceBoundPSM.address,
+    usdcPriceBoundPSM.address
+  );
+  await migratorRouter.deployed();
+
+  console.log(`\nMigrator Router deployed to: ${migratorRouter.address}`);
+
+  return {
+    // put returned contract objects here
+  };
+};
+
+// Do any setup necessary for running the test.
+// This could include setting up Hardhat to impersonate accounts,
+// ensuring contracts have a specific state, etc.
+const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in setup for vip${vipNumber}`);
+};
+
+// Tears down any changes made in setup() that need to be
+// cleaned up before doing any validation checks.
+const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in teardown for vip${vipNumber}`);
+};
+
+// Run any validations required on the vip using mocha or console logging
+// IE check balances, check state of contracts, etc.
+const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  console.log(`No actions to complete in validate for vip${vipNumber}`);
```

Comment:
Please add validations here on the following:

- TempCoreRef worked correctly and the Volt token address is correct on the deployed PSM's
- PSM params were properly set
- Migrator contract has correct params
- Router contract has correct params
- ERC20 Allocator has successfully removed the previous PSM's and their links to deposits, and that the new PSM's have been connected and linked to these PCV Deposits.